Digipost/src/main/java/no/digipost/android/gui/metadata/ExternalLinkWebview.java

```diff
@@ -0,0 +1,233 @@
+        String hostUrl = "";
+        try {
+            hostUrl = new URL(fileUrl).getHost();
+        }catch (MalformedURLException e){
+            hostUrl = fileUrl;
+        }
+
+        String title = getString(R.string.externallink_download_title);
+        String message = format(getString(R.string.externallink_download_message), fileName, hostUrl);
+
+        AlertDialog.Builder builder = DialogUtitities.getAlertDialogBuilderWithMessageAndTitle(this, message, title);
+        builder.setPositiveButton(getString(R.string.externallink_download_title), new DialogInterface.OnClickListener() {
+            public void onClick(final DialogInterface dialog, final int id) {
+                downloadFile(userAgent, content, mimeType, contentLength);
+                dialog.dismiss();
+            }
+        }).setCancelable(false).setNegativeButton(getString(R.string.abort), new DialogInterface.OnClickListener() {
+            public void onClick(final DialogInterface dialog, final int id) {
+                dialog.cancel();
+                finish();
+            }
+        });
+
+        builder.create().show();
+    }
+
+    private void showPermissionsDialog() {
+
+        String message = getString(R.string.externallink_permissions);
+        final Activity activity = this;
+        AlertDialog.Builder builder = DialogUtitities.getAlertDialogBuilderWithMessage(activity, message);
+        builder.setPositiveButton(getString(R.string.ok), new DialogInterface.OnClickListener() {
+            public void onClick(final DialogInterface dialog, final int id) {
+                Permissions.requestWritePermissionsIfMissing(activity, activity);
+                dialog.dismiss();
+            }
+        }).setCancelable(false).setNegativeButton(getString(R.string.abort), new DialogInterface.OnClickListener() {
+            public void onClick(final DialogInterface dialog, final int id) {
+                dialog.cancel();
+                finish();
+            }
+        });
+
+        builder.create().show();
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        webView.loadUrl(fileUrl);
+    }
+
+    private void showMissingPermissionsDialog(){
+        String message = getString(R.string.externallink_permissions);
+        AlertDialog.Builder builder = DialogUtitities.getAlertDialogBuilderWithMessage(this, message);
+        builder.setPositiveButton(getString(R.string.ok), new DialogInterface.OnClickListener() {
+            public void onClick(final DialogInterface dialog, final int id) {
+                dialog.dismiss();
+                finish();
+            }
+        });
+        builder.create().show();
+    }
+
+    private void downloadFile(final String userAgent, final String content, final String mimeType, final long contentLength) {
```

Comment:
Er dette noe det går an å dra ut i en slags convenience-fil / klasse?

Reply:
Godt poeng! Ville ha gjort det om den skal brukes flere steder, men akkurat denne skal kun brukes i dette viewet, så vil nok forvirre mer om den er andre steder også. For den er litt spesiell :)